### PR TITLE
Support basic error handling

### PR DESCRIPTION
**Pull request by shendiaomo**
Now we'll `panic` if an exception happens. In this PR only `torch.MM` is processed this way. Later I'll rewrite other functions

### CODE REVIEW

File: cgotorch/cgotorch.h [hunk 1]
```diff
@@ -9,8 +9,12 @@ extern "C" {
 
 typedef void *Tensor;
 typedef void *Optimizer;
+typedef struct {
+  Tensor t;
+  char *err;
+} Result;
 Tensor RandN(int rows, int cols, int require_grad);
-Tensor MM(Tensor a, Tensor b);
+Result MM(Tensor a, Tensor b);
```
Comment (wangkuiyi) [inline cgotorch/cgotorch.h:17]: I think the C-conventional way to return more than one values is to have an additional parameter, other than defining `Result` type.

```c
Tensor MM(Tensor a, Tensor b, char** err);
```

Comment (shendiaomo) [reply cgotorch/cgotorch.h]: Done.

File: cgotorch/cgotorch.cc [hunk 2]
```diff
@@ -16,10 +16,18 @@ Tensor RandN(int rows, int cols, int require_grad) {
...
-  return new at::Tensor(c);
+char *MM(Tensor a, Tensor b, Tensor *result) {
+  try {
+    at::Tensor c =
+        at::mm(*static_cast<at::Tensor *>(a), *static_cast<at::Tensor *>(b));
+    *result = new at::Tensor(c);
+    return nullptr;
+  } catch (const std::exception& e) {
+    auto len = strlen(e.what());
+    auto r = new char[len + 1];
```
Comment (wangkuiyi) [inline cgotorch/cgotorch.cc:27]: So every time we call MM, we need to call FreeString() to release the error right?  I don't see the call to FreeString in the examples?

File: cgotorch/Makefile [hunk 3]
```diff
@@ -7,7 +7,7 @@ macos/libtorch : libtorch-macos-1.5.1.zip
 	unzip -qq -o $< -d macos
 
 libcgotorch.so : cgotorch.cc cgotorch.h macos/libtorch
-	rm -f libtorch
+	rm -rf libtorch/
```
[the remark this answers is not in the document]
Comment (wangkuiyi) [reply cgotorch/Makefile]: I fixed it.

File: cgotorch/cgotorch.h [hunk 4]
```diff
@@ -9,8 +9,12 @@ extern "C" {
 
 typedef void *Tensor;
 typedef void *Optimizer;
+typedef struct {
+  Tensor t;
+  char *err;
+} Result;
```
[the remark this answers is not in the document]
Comment (wangkuiyi) [reply cgotorch/cgotorch.h]: I fixed it.

**Pull request merged**